telegram/src/main/kotlin/com/github/kotlintelegrambot/entities/inputmedia/InputMedia.kt

```diff
@@ -1,7 +1,112 @@
 package com.github.kotlintelegrambot.entities.inputmedia
 
-abstract class InputMedia {
+import com.github.kotlintelegrambot.entities.TelegramFile
+import com.google.gson.annotations.SerializedName
+
+/**
+ * Represents the content of a media message to be sent.
+ * https://core.telegram.org/bots/api#inputmedia
+ */
+sealed class InputMedia {
     abstract val type: String
-    abstract val media: String
+    abstract val media: TelegramFile
     abstract val caption: String?
+    abstract val parseMode: String?
+}
+
+/**
+ * Interface to mark all the media types that can be sent within a group of media for
+ * operations like `sendMediaGroup`.
+ */
+interface GroupableMedia
+
+class MediaGroup private constructor(val medias: Array<out GroupableMedia>) {
```

Comment:
a Set instead of Array could helps here?

Reply:
hmm... the point is that you can send the same media two times, I guess that's not the usual use case, but using a set wouldn't be accurate enough for this reason. 

Also, using an `Array` lets us taking advantage of `varargs` in an easier way